### ISSUE TITLE
Installation

### ISSUE THREAD

**Issue by ted-miller** [gavanderhoorn/micro_xrce_dds_ros_idl_dissectors#1]
I've never really had any trouble loading .lua scripts into wireshark. But these scripts do not like me.

Can you clarify the relative paths of these files? Is this correct:

```
\plugins
| - \ros_idl_types
| -  | - (specific message types)
| - dds-xrce-idl
| - dds-xrce-proto
| - dds-xrce-types
| - ros_idl_type_map
```

On Windows, I dump everything in, then `builtin_interfaces_msg_Duration` complains that it can't find `dds-xrce-idl`. Then it lists a bunch of unexpected search paths.

![image](https://user-images.githubusercontent.com/4623811/202014918-da76da39-32a3-4fe7-a40c-b6b22b3d4df1.png)

So, I moved the `dds-xrce-___` files to be on that search path. Then it complains that it can't find `builtin_interfaces_msg_Time`, which is in the same dir as the file throwing the error.
![image](https://user-images.githubusercontent.com/4623811/202015215-c469a5bc-154c-46c1-ac1c-dfb40cb8e46e.png)

So, I tried putting everything in `Wireshark` directory. Now I don't get an error, but I don't think it's loading the .lua scripts. It's certainly not dissecting the packets.

------------------------

So, let's assume it's a Windows thing. I move over to my Linux box. I `apt install wireshark`. Then I navigate to the `~/.local/lib` folder. There is no Wireshark subdirectory. I create the `wireshark/plugins` directory and paste in the .lua files, using the directory structure shown at the top of this post. I open the capture, and it just shows a bunch of raw UDP.

I can manually tell it to decode as DDS-XRCE, but it's not doing anything for the payload. I can't find another decode option for "ros_idl_types".

I also tried pasting these files in `/usr/share/wireshark/plugins`. Same result. 

-----------------------

-----------------------------

I'm guessing that it has something to do with the `require` keyword in the scripts. In the HSES dissector, I used the `dofile` (based on someone else's work).

```
local folderOfThisFile = script_path():match("@?(.*[/\\])"):gsub("\\", "/") -- get folder name and replace backslash to slash

dofile( folderOfThisFile.. "wiresharklib/hses_name.lua" )
dofile( folderOfThisFile.. "wiresharklib/hses_header.lua" )
dofile( folderOfThisFile.. "wiresharklib/hses_baseCommands.lua" )
```

**Comment by gavanderhoorn** [comment]
Yeah, there seem to be some differences between Windows and Linux, and different Wireshark versions.

The current implementation works on my main development machine, but won't work on another. I'll see if I can sort things out.

re: location of the files: everything should work by just copying the files into your regular `plugins` location. No need to copy it anywhere else.

**Comment by gavanderhoorn** [comment]
I realise this is really old, but 17e00387fab0af39fc8eb0a4efe98c664b3a9925 may have fixed this.

No guarantees as I haven't tested it on Windows.

**Comment by ted-miller** [comment]
On Windows:

```
Lua: Error during loading:
C:\Program Files\Wireshark\plugins\4.6\ros_idl_type_map.lua:6: attempt to concatenate a nil value (local 'dir')
stack traceback:
	C:\Program Files\Wireshark\plugins\4.6\ros_idl_type_map.lua:6: in main chunk
	[C]: in function 'require'
```

ros_idl_type_map.lua:
https://github.com/gavanderhoorn/micro_xrce_dds_ros_idl_dissectors/blob/17e00387fab0af39fc8eb0a4efe98c664b3a9925/ros_idl_type_map.lua#L5-L6

**Comment by ted-miller** [comment]
Gemini found the problem. Windows doesn't like the forward slash. Change line 5 to:

`local dir = debug.getinfo(1, "S").source:sub(2):match("^(.*[/\\])")`
